tests/collect_all_nested/BUILD.out

```diff
@@ -0,0 +1,22 @@
+load("@aspect_rules_js//js:defs.bzl", "js_library")
+load("@aspect_rules_ts//ts:defs.bzl", "ts_project")
+
+# gazelle:js_root
+# gazelle:js_jest_config :jest.config
+# gazelle:js_quiet
+
+ts_project(
+    name = "index",
+    srcs = ["index.ts"],
+    deps = ["//my_module"],
+)
+
+ts_project(
+    name = "some_other_file",
+    srcs = ["some_other_file.ts"],
```

Comment:
why is this not part of the `index` target

Reply:
index does not import `some_other_file` this was a change we made a while ago to make `js_collect_barrels` less likely to cause cyclical dependencies